rock_paper_scissors/api/models.py

```diff
@@ -0,0 +1,65 @@
+"""
+Relationships with Foreign Keys, esp. relationship() and FK values
+Why do we need to establish columns for players in game?
+difference with many to one and one to many -- if you back_populate either relationship type,
+    does it matter which you use?
+
+"""
+
+from sqlalchemy import Column, Integer, String
+from sqlalchemy import Enum
+from sqlalchemy import ForeignKey
+from sqlalchemy.ext.declarative import as_declarative
+from sqlalchemy.orm import relationship
+
+
+@as_declarative()
+class ModelBase(object):
+    def __tablename__(self):
+        return self.__class__.__name__.lower()
+
+
+class Game(ModelBase):
+    id = Column(Integer, primary_key=True)
+    player_1_id = Column(Integer, ForeignKey('user.id'))
+    player_2_id = Column(Integer, ForeignKey('user.id'))
+    round_id = Column(Integer, ForeignKey('round.id'))
+    rounds_choice = Column(Integer)
+    rounds_decrement = Column(Integer)
+    current_round = Column(Integer)
+    winner = Column(String, ForeignKey('user.id'))
+
+    round = relationship('Round', back_populates='game')
+    user = relationship('User', back_populates='game', uselist=False)
```

Comment:
Thanks the explanation, that clears up the understanding about relationships a bit more.

I have a few questions that I've been thinking about, basically I think my lack of understanding with the  `Game` model in its' relation to the others models could be addressed by understanding:

1. When you are building models, do you think about the relations of models to models such as one-to-many, many-to-one, one-to-one, many-to-many when you are designing? 
2. When you use a foreign key in a model, is the entire instance/object that was referenced with the foreign key also accessible to that model with the foreign key?
i.e if `Game` has `User.id` reference, can we access the other values (`User.name, User.score, User.choices`) for `User.id`?
```
class Game(ModelBase):
    ...
    player_1_id = Column(Integer, ForeignKey('user.id'))
```
```
class User(ModelBase):
    ...
    id = Column(Integer, primary_key=True)
    name = Column(String)
    score = Column(Integer)
    choices = Column(String, ForeignKey('choice.move'))

    game = relationship('Game', back_populates='user')
    choice = relationship('Choice', back_populates='user')
    round = relationship('Round', back_populate='user')
```

3. If a bi-directional relationship is needed to be established between 2 tables, how do you establish the relationship without using `back_populate`/ `back_ref`?
4. Why is it that you use all the attributes referenced for the foreign_keys as a list? ie `foreign_keys=[referenced attribute]`?

 I was reading the documentation on foreign_keys arguments for relationships and came across [this](http://docs.sqlalchemy.org/en/latest/orm/join_conditions.html#handling-multiple-join-paths)

```
billing_address = relationship("Address", foreign_keys="[Customer.billing_address_id]")

billing_address = relationship("Address", foreign_keys="Customer.billing_address_id")
```
Is the string version for only one value reference? If it is, can we still wrap it in a list?